### PR TITLE
[repo] Update Microsoft.Extensions.* packages

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Update Microsoft.Extensions.* packages

## Merge requirement checklist

* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -16,23 +16,24 @@
...
-          Microsoft.Extensions.Logging even during major version bumps, so compatibility is not a concern here.
+          these packages even during major version bumps, so compatibility is not a concern here.
     -->
-    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.2.23479.6" />
 
+    <PackageVersion Include="Microsoft.Extensions.Configuration" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.2.23479.6" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Configuration" Version="[3.1.0,)" />
```
Comment (utpilla) [inline Directory.Packages.props:34]: This package will be updated in a different PR. It's not as straightforward as simply updating this package breaks the build.

**Review by Kielek — APPROVED** [review]
:(

File: Directory.Packages.props [hunk 2]
```diff
@@ -16,23 +16,24 @@
...
         3) Since version 3.1.0, the .NET runtime team is holding a high bar for backward compatibility on
-          Microsoft.Extensions.Logging even during major version bumps, so compatibility is not a concern here.
+          these packages even during major version bumps, so compatibility is not a concern here.
     -->
-    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.2.23479.6" />
 
+    <PackageVersion Include="Microsoft.Extensions.Configuration" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="8.0.0-rc.2.23479.6" />
+    <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0-rc.2.23479.6" />
```
Comment (CodeBlanch) [inline Directory.Packages.props:33]: If we're going to keep all of these in sync could we introduce a new property like `<MicrosoftExtensionsPackageDefaultVersion>8.0.0-rc.2.23479.6</MicrosoftExtensionsPackageDefaultVersion>` they could all use?

Comment (utpilla) [reply Directory.Packages.props:33]: I would prefer each of them having the version defined explicitly instead of introducing a layer of indirection in determining which version of the package is being used.

Also, in future if only a few of them get patches then we would need lesser changes as we would be able to update the versions of only the affected packages.

Comment (TimothyMothra) [reply Directory.Packages.props:33]: The indirection might also prevent tools like Dependabot from auto-updating packages.

Comment (CodeBlanch) [reply Directory.Packages.props:33]: Fair enough! Just FYI we did confirm with @tarekgh that MSE packages can be versioned independently.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: Directory.Packages.props (hunk 1 above)
Comment (utpilla) [reply Directory.Packages.props]: This was covered in #5020

**Review by CodeBlanch — APPROVED** [review]
LGTM
